### PR TITLE
Add custom RPCs to creator staking

### PR DESCRIPTION
**Pull request by F3Joule**

### CODE REVIEW

File: pallets/creator-staking/src/functions.rs [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+                continue;
+            }
+
+            loop {
+                let (era, _) = stakes_info.claim();
+                if era >= unregistered_era || era == 0 {
+                    break;
+                }
+
+                available_claims_by_creator.entry(creator).and_modify(|e| *e += 1).or_insert(1);
```
Comment (olehmell) [inline pallets/creator-staking/src/functions.rs:407]: So here we are calculating how many times/eras we can claim. So if we have [(1, 5), (2, 10)] on response, on UI we will put claim for the creator #1 - 5 times and for #2 - 10 times, right?

Comment (olehmell) [reply pallets/creator-staking/src/functions.rs:407]: Pretty hard understand from name what does mean the `u32` in response. So maybe "available_claim_count_by_creator"

File: pallets/creator-staking/rpc/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+/// Provides RPC method to query a domain price.
+pub struct CreatorStaking<C, P> {
+    /// Shared reference to the client.
+    client: Arc<C>,
+    _marker: std::marker::PhantomData<P>,
+}
+
+impl<C, P> CreatorStaking<C, P> {
+    /// Creates a new instance of the Domains Rpc helper.
```
Comment (olehmell) [inline pallets/creator-staking/rpc/src/lib.rs:56]:
```suggestion
    /// Creates a new instance of the CreatorStaking Rpc helper.
```

File: pallets/creator-staking/src/functions.rs [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+
+        for creator_id in target_creators {
+            let mut staker_info_for_creator = Self::staker_info(&staker, creator_id);
+
+            let mut unregistered_era = current_era;
+            if let Some(creator_info) = Self::registered_creator(creator_id) {
+                if let CreatorState::Unregistered(era) = creator_info.state {
+                    unregistered_era = era;
+                }
+            }
```
Comment (olehmell) [inline pallets/creator-staking/src/functions.rs:342]: Same block code on 392-397 lines. Not sure that we can optimize it, but wanted to notify

File: pallets/creator-staking/src/functions.rs [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+
+            if staker_info_for_creator.stakes.is_empty() {
+                estimated_rewards.push((creator_id, Zero::zero()));
+                continue;
+            }
+
+            let mut total_staker_rewards_for_eras: BalanceOf<T> = Zero::zero();
+            loop {
+                let (era, staked) = staker_info_for_creator.claim();
+                if era >= unregistered_era || era == 0 {
```
Comment (olehmell) [inline pallets/creator-staking/src/functions.rs:352]: You are using same condition in `available_claims_by_staker`, so maybe make sense to extract it

File: pallets/creator-staking/src/functions.rs [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+                    break;
+                }
+                let creator_stake_info = Self::creator_stake_info(creator_id, era).unwrap_or_default();
+
+                total_staker_rewards_for_eras = total_staker_rewards_for_eras.saturating_add(
+                    Self::calculate_reward_for_staker_in_era(&creator_stake_info, staked, era)
```
Comment (olehmell) [inline pallets/creator-staking/src/functions.rs:358]: if we have `default` value for `creator_stake_info` here, does make sense to call calculate function anyway?

**Review by olehmell — CHANGES_REQUESTED** [review]

File: pallets/creator-staking/rpc/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+
+    #[method(name = "posts_availableClaimsByStaker")]
+    fn available_claims_by_staker(
+        &self,
+        staker: AccountId,
+        at: Option<BlockHash>,
+    ) -> RpcResult<Vec<(SpaceId, u32)>>;
+}
+
+/// Provides RPC method to query a domain price.
```
Comment (olehmell) [inline pallets/creator-staking/rpc/src/lib.rs:48]: Update comment

File: pallets/creator-staking/rpc/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,142 @@
+//! RPC interface for the domains pallet.
```
Comment (olehmell) [inline pallets/creator-staking/rpc/src/lib.rs:1]:
```suggestion
//! RPC interface for the creator-staking pallet.
```

File: pallets/creator-staking/Cargo.toml [hunk 8]
```diff
@@ -0,0 +1,63 @@
+[package]
+name = "pallet-creator-staking"
+version = "0.1.8"
```
[the remark this answers is not in the document]
Comment (F3Joule) [reply pallets/creator-staking/Cargo.toml:3]: Yeah, good for now. We'll bump crates version to 0.1.9 because of this RPC.

**Pull request merged**